src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1477,7 +1477,9 @@ private string GetLocalDebugName(ILocalSymbolInternal local, out LocalDebugId lo
             // only user-defined locals should be named during lowering:
             Debug.Assert((local.Name == null) == (localKind != SynthesizedLocalKind.UserDefined));
 
-            if (!localKind.IsLongLived())
+            // Skip naming locals generated by instrumentation as they might not have a local syntax offset.
+            // Locals generated by instrumentation might exist in methods which do not contain a body (auto property initializers).
+            if (!localKind.IsLongLived() || localKind == SynthesizedLocalKind.InstrumentationPayload)
             {
```

Comment:
This breaks invariant that is required to keep EnC working. Is there any value in instrumenting compiler generated code?